plugin/pulpcore/plugin/download/asyncio/exceptions.py

```diff
@@ -0,0 +1,34 @@
+from gettext import gettext as _
+
+from pulpcore.exceptions import PulpException
+
+
+class DownloaderValidationError(PulpException):
```

Comment:
Considering your change in tasking to log non-fatal `PulpException` with stack traces - these exceptions extending `PulpException` will cause validation exceptions to be logged with a stack trace.  This seems undesirable.  I think we only want to log stack traces when unhandled exceptions indicate something is broken.

Reply:
The downloaders can raise these exceptions, but they are designed to be caught by the user and not to propogate all the way up. For direct usage of the downloaders the user will need to catch and handle these types of exceptions. I wrote some about that [here](https://github.com/pulp/pulp/pull/3129/files#diff-a947592032b960576ac9523a9a255e67R97).

In the event that the exception isn't caught, I think having a stack trace is what we want. Celery typically does log the stack trace of a fatal exception, but for some reason it wasn't logging PulpException based exceptions. I think we want to also log those stacktraces since they are fatal task exceptions and to be consistent with celery's logging. What do you think given all ^?